### PR TITLE
OSSM-3701 Refactor TestSMCPAddons (T34)

### PR DESCRIPTION
**Pull request by fjglira**
Jira: https://issues.redhat.com/browse/OSSM-3701

Refactor the test case TestSMCPAddons, this test case only contains now a sub-test case to check if a validation error is returned when is tried to enable the 3scale addon in 2.1 or newer version. Changes made:

- Cleanup: recreate the namespace
- Rename the func MergePatch for only Patch and added a temporary "hack" to add || true at the end of the patch to not get an error when the patch return a validation error
- The test is setting the patch and checking that we get a validation error with the string: `support for 3scale has been removed`
- Added TODO in func to remind that is a temporary solution add ||true as an additional params is needed

NOTE: Do not merge until https://github.com/maistra/maistra-test-tool/pull/405 is merged because this branch is forked from that branch (Depends from that changes)

### CODE REVIEW

File: pkg/tests/ossm/smcp_addons_test.go [hunk 1]
```diff
@@ -15,29 +15,36 @@
...
-		}
+		// Created a subtest because we need to add more test related to Addons in the future.
+		t.NewSubTest("3scale_addon").Run(func(t TestHelper) {
+			t.LogStep("Enable 3scale in a SMCP expecting to get validation error.")
+			oc.Patch(t,
+				meshNamespace,
+				fmt.Sprintf("smcp/%s", smcpName),
+				"merge",
+				`{"spec":{"addons":{"3scale":{"enabled":true}}}}`, // TODO: remove || true when we modify the func to detect the error expected.
+				"|| true", //TODO: hack, remove this when we modify the func to detect the error expected. This is to avoid the test to fail because the patch is not applied.
```
[the remark this answers is not in the document]
Comment (fjglira) [reply pkg/tests/ossm/smcp_addons_test.go]: Done

File: pkg/util/oc/pod_commands.go [hunk 2]
```diff
@@ -26,7 +26,8 @@ func Exec(t test.TestHelper, podLocator PodLocatorFunc, container string, cmd st
 		checks...)
 }
 
-func MergePatch(t test.TestHelper, ns string, rs string, ptype string, patch string, checks ...assert.CheckFunc) {
+// TODO: avoidError is a hack to avoid error when patching a resource that doesn't exist
+func Patch(t test.TestHelper, ns string, rs string, ptype string, patch string, checks ...assert.CheckFunc) {
```
Comment (yxun) [inline pkg/util/oc/pod_commands.go:30]: I see this line 30 has been updated in maistra-2.4 branch.
have you fetched and synced the maistra-2.4 branch change ?
https://github.com/maistra/maistra-test-tool/blob/maistra-2.4/pkg/util/oc/pod_commands.go#L30 

It's strange and not clear why this diff is still in your PR.

Comment (fjglira) [reply pkg/util/oc/pod_commands.go]: Please check again, I fix some merge conflicts

**Comment by luksa** [comment]
@fjglira I can't review this because the commits are now a mess now that we've merged your other PR. Please ping me when you rebase.

**Comment by fjglira** [comment]
Deleting this PR and creating a new one

**Pull request closed without merging**